packages/dds/matrix/src/test/time/sharedMatrix.spec.ts

```diff
@@ -40,242 +43,813 @@ describe("SharedMatrix execution time", () => {
+				});
 
-							// Setup
+				// Test the execute time of the SharedMatrix for setting a string in a cell for a given number of times.
+				describe(`Cell Value Setting`, () => {
+					benchmark({
+						title: `Set a 3-character string in ${count} cells`,
+						before: async () => {
 							localMatrix = createLocalMatrix({
 								id: "testLocalMatrix",
 								size: matrixSize,
 								initialValue: matrixValue,
 							});
-
-							// Operation
-							const before = state.timer.now();
+						},
+						benchmarkFn: () => {
 							for (let i = 0; i < count; i++) {
-								localMatrix.removeCols(Math.floor(localMatrix.colCount / 2), 1);
-								localMatrix.removeRows(Math.floor(localMatrix.rowCount / 2), 1);
+								localMatrix.setCell(i, i, "abc");
 							}
-							const after = state.timer.now();
-
-							// Measure
-							duration = state.timer.toSeconds(before, after);
-
-							// Collect data
-						} while (state.recordBatch(duration));
-					},
-					// Force batch size of 1
-					minBatchDurationSeconds: 0,
-					// Matrix size 100 benchmarks use increased duration (10s) to improve statistical significance.
-					maxBenchmarkDurationSeconds: matrixSize === 100 ? 10 : 5,
-				});
-
-				// Test the execute time of the SharedMatrix for setting a string in a cell for a given number of times.
-				benchmark({
-					title: `Set a 3-character string in ${count} cells`,
-					before: async () => {
-						localMatrix = createLocalMatrix({
-							id: "testLocalMatrix",
-							size: matrixSize,
-							initialValue: matrixValue,
-						});
-					},
-					benchmarkFn: () => {
-						for (let i = 0; i < count; i++) {
-							localMatrix.setCell(i, i, "abc");
-						}
-					},
+						},
+					});
+
+					benchmark({
+						type: BenchmarkType.Measurement,
+						title: `Undo set a 3-character string in ${count} cells`,
+						benchmarkFnCustom: async <T>(state: BenchmarkTimer<T>) => {
+							let duration: number;
+
+							do {
+								// Since this setup one collects data from one iteration, assert that this is what is expected.
+								assert.equal(state.iterationsPerBatch, 1);
+
+								// Setup
+								localMatrix = createLocalMatrix({
+									id: "testLocalMatrix",
+									size: matrixSize,
+									initialValue: matrixValue,
+								});
+								undoStack = new UndoRedoStackManager();
+								localMatrix.openUndo(undoStack);
+								for (let i = 0; i < count; i++) {
+									localMatrix.setCell(i, i, "abc");
+								}
+
+								// Operation
+								const before = state.timer.now();
+								for (let i = 0; i < count; i++) {
+									undoStack.undoOperation();
+								}
+								const after = state.timer.now();
+
+								// Measure
+								duration = state.timer.toSeconds(before, after);
+
+								// Collect data
+							} while (state.recordBatch(duration));
+						},
+						// Force batch size of 1
+						minBatchDurationSeconds: 0,
+						maxBenchmarkDurationSeconds: matrixSize === 100 ? 10 : 5,
```

Comment:
This appears to be the same for all of the tests. Let's extract it into a variable and document it.

Reply:
Updated